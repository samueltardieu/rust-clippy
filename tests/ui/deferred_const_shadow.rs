#![warn(clippy::deferred_const_shadow)]
#![allow(unused)]

use std::path::MAIN_SEPARATOR;
const MESSAGE: &str = "ðŸ¦€";

fn main() {
    println!("{MESSAGE}");
    //~^ ERROR: using a shadowing constant defined later in the same scope
    const MESSAGE: &str = "Hello Rust!";

    // No lint here, as this appears after the constant definition
    println!("{MESSAGE}");

    // No lint here, as no constant with the same name is shadowed
    println!("{OTHER_MESSAGE}");
    const OTHER_MESSAGE: &str = "foobar";

    // No lint (because no shadow)
    println!("{MAIN_SEPARATOR}");
    const MAIN_SEPARATOR: &str = "::";
}
