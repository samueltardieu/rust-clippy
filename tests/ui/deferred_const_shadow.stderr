error: using a shadowing constant defined later in the same scope
  --> tests/ui/deferred_const_shadow.rs:8:16
   |
LL |     println!("{MESSAGE}");
   |                ^^^^^^^
   |
note: this constant is shadowed in the current scope and will not be used
  --> tests/ui/deferred_const_shadow.rs:5:1
   |
LL | const MESSAGE: &str = "ðŸ¦€";
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: this constant is used earlier in the scope and should be declared before it is referenced
  --> tests/ui/deferred_const_shadow.rs:10:5
   |
LL |     const MESSAGE: &str = "Hello Rust!";
   |     ^^^^^^^^^^^^^^^^^^^
   = note: `-D clippy::deferred-const-shadow` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::deferred_const_shadow)]`

error: aborting due to 1 previous error

